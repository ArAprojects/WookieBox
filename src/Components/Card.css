@font-face {
    font-family: 'distant_galaxyregular';
    src: url('/../distgrg_-webfont.woff2') format('woff2'),
         url('/../distgrg_-webfont.woff') format('woff');
    font-weight: normal;
    font-style: normal;
  }

.Card {
    height: 350px;
    width: 20%;
    border-radius: 15px;
    background-color: lightgrey;
    opacity: 0.7;
    text-align: center;
    margin-bottom: 40px;
    margin-left: 5px;
    margin-right: 5px;
    justify-content: space-evenly;
    overflow: scroll;
}

p {
  font-size: 1.8em;
  margin: 10px;
}


.name-and-button {
  display: flex;
  justify-content: center;
}

.cardContainer {
    display: flex;
    justify-content: space-evenly;
    flex-flow: row wrap;
  }

  .favorite {
    background: url(/../star.png) no-repeat;
  }

  img {
    height: 35px;
  }

  hr {
    border: 2px solid black;
  }