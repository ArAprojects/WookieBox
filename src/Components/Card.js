import React from 'react'
// import { Link } from 'react-router-dom'
import './Card.css'
const uuidv4 = require('uuid/v4');

const Card = ( {data, toggleFavorite, favoriteStatus} ) => {
    const displayCards = data.map(card => {
      const { name, population, species, homeworld, residents, terrain, climate, model, vehicle_class, passengers } = card;
        const notDefined = (keys, type) => {
          if(keys === undefined) {
            return ''
          }
          else if (Array.isArray(keys)) {
            return keys.map(key => key[type])
          }
          else {
            return keys[type]
          }
        }


      return (
        <section className="Card" key={uuidv4()}>
          <div className="name-and-button">
            <p>{name}</p>
             <button onClick={() => toggleFavorite(card) }>
               <h5>
                 {card.favorite ? <img src="/../fav_star.png" alt="fav-star"/> : <img src="/../star.png" alt="star-vector"/>}
                </h5>
              </button>
          </div>
          <div className="content">
            {terrain && <p>Terrain: {terrain}</p>}
            {climate && <p>Climate: {climate}</p>}
            {model && <p>Model: {model}</p>}
            {vehicle_class && <p>Class: {vehicle_class}</p>}
            {passengers && <p>Passenger Capacity: {passengers}</p>}
            {notDefined(species, "name") && <p>Species: {notDefined(species, "name")}</p>}
            {notDefined(homeworld, "name") && <p>Homeworld: {notDefined(homeworld, "name")}</p>}
            {notDefined(species, "language") && <p>Language: {notDefined(species, "language")}</p>}
            {population && <p>Population: {population}</p>}
            {notDefined(residents, "name") && <p>Residents: {notDefined(residents, "name")}</p>}
          </div>
        </section>
      )
    });

    return (
    <div className="cardContainer">
      {displayCards}
    </div>
  );
}

export default Card
