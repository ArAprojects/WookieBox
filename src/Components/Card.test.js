import React from 'react'
import { shallow } from 'enzyme'
import Card from './Card'

describe('Card', () => {
    it('should return a snapshot of a card object with people data', () => {
        const data = [{key: "", id: 1, name: "Luke Skywalker", toggleFavorite: jest.fn(), homeworld: "Tatooine", species: "Human", language: "Galactic Basic"}]
        const wrapper = shallow(
            <Card
            data={data}
            key={data.key}
            id={data.id}
            name={data.name}
            toggleFavorite={data.toggleFavorite}
            homeworld={data.homeworld}
            species={data.species}
            language={data.language}
            terrain={data.terrain}
            population={data.population}
            climate={data.climate}
            residents={data.residents}
            model={data.model}
            vehicle_class={data.vehicle_class}
            passengers={data.passengers}
            />)

        expect(wrapper).toMatchSnapshot()
    })

    it('should return a snapshot of a card object with planet data', () => {
        const data = [{key: "", id: 7, name: "Dagobah", toggleFavorite: jest.fn(), terrain: "swamp, jungles", climate: "murky", population: "unknown", residents: []}]
        const wrapper = shallow(
            <Card
            data={data}
            key={data.key}
            id={data.id}
            name={data.name}
            toggleFavorite={data.toggleFavorite}
            homeworld={data.homeworld}
            species={data.species}
            language={data.language}
            terrain={data.terrain}
            population={data.population}
            climate={data.climate}
            residents={data.residents}
            model={data.model}
            vehicle_class={data.vehicle_class}
            passengers={data.passengers}
            />)

        expect(wrapper).toMatchSnapshot()
    })

    it('should return a snapshot of a card object with planet data', () => {
        const data = [{key: "", id: 7, name: "Dagobah", toggleFavorite: jest.fn(), terrain: "swamp, jungles", climate: "murky", population: "unknown", residents: []}]
        const wrapper = shallow(
            <Card
            data={data}
            key={data.key}
            id={data.id}
            name={data.name}
            toggleFavorite={data.toggleFavorite}
            homeworld={data.homeworld}
            species={data.species}
            language={data.language}
            terrain={data.terrain}
            population={data.population}
            climate={data.climate}
            residents={data.residents}
            model={data.model}
            vehicle_class={data.vehicle_class}
            passengers={data.passengers}
            />)

        expect(wrapper).toMatchSnapshot()
    })
})
