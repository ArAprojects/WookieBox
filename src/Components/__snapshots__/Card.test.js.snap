// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should return a snapshot of a card object with people data 1`] = `
<div
  className="cardContainer"
>
  <section
    className="Card"
    key="d2da649f-60d6-4d68-9e98-2b37558e361c"
  >
    <div
      className="name-and-button"
    >
      <p>
        Luke Skywalker
      </p>
      <button
        onClick={[Function]}
      >
        <h5>
          X
        </h5>
      </button>
    </div>
    <div
      className="content"
    />
  </section>
</div>
`;

exports[`Card should return a snapshot of a card object with planet data 1`] = `
<div
  className="cardContainer"
>
  <section
    className="Card"
    key="49778777-de8e-470c-92c8-809134d87c58"
  >
    <div
      className="name-and-button"
    >
      <p>
        Dagobah
      </p>
      <button
        onClick={[Function]}
      >
        <h5>
          X
        </h5>
      </button>
    </div>
    <div
      className="content"
    >
      <p>
        Terrain: 
        swamp, jungles
      </p>
      <p>
        Climate: 
        murky
      </p>
      <p>
        Population: 
        unknown
      </p>
      <p>
        Residents: 
      </p>
    </div>
  </section>
</div>
`;

exports[`Card should return a snapshot of a card object with planet data 2`] = `
<div
  className="cardContainer"
>
  <section
    className="Card"
    key="b0b63cec-d3d6-405d-bb04-4f4b49da59b8"
  >
    <div
      className="name-and-button"
    >
      <p>
        Dagobah
      </p>
      <button
        onClick={[Function]}
      >
        <h5>
          X
        </h5>
      </button>
    </div>
    <div
      className="content"
    >
      <p>
        Terrain: 
        swamp, jungles
      </p>
      <p>
        Climate: 
        murky
      </p>
      <p>
        Population: 
        unknown
      </p>
      <p>
        Residents: 
      </p>
    </div>
  </section>
</div>
`;
